weekendhomework6/src/ba/bitcamp/weekendhomework/GUI/tasks/Task4.java

```diff
@@ -0,0 +1,101 @@
+package ba.bitcamp.weekendhomework.GUI.tasks;
+
+import java.awt.BorderLayout;
+import java.awt.event.ActionEvent;
+import java.awt.event.ActionListener;
+import javax.swing.JButton;
+import javax.swing.JFrame;
+import javax.swing.JLabel;
+
+public class Task4 extends JFrame {
+
+	/**
+	 * 
+	 */
+	private static final long serialVersionUID = -3808415954214927694L;
+
+	public int num;
+	private String line = "";
+	private String[] characters = { "A", "B", "C", "D", "E", "F", "K", "L",
+			"M", "N", "U", "V", "Z" };
+
+	private JButton button1 = new JButton(" > ");
+	private JButton button2 = new JButton(" < ");
+	private JButton button3 = new JButton(" Insert ");
+	private JButton button4 = new JButton(" Character: ");
+	private JLabel text = new JLabel("_");
+
+	public Task4() {
+
+		setVisible(true);
+		setResizable(false);
+		setLocation(350, 350);
+		setSize(400, 200);
+
+		setTitle(" Task 2 ");
+
+		setLayout(new BorderLayout());
+		add(button1, BorderLayout.WEST);
+		add(button2, BorderLayout.EAST);
+		add(button3, BorderLayout.SOUTH);
+		add(button4, BorderLayout.NORTH);
+		add(text, BorderLayout.CENTER);
+
+		button1.addActionListener(new Action());
+		button2.addActionListener(new Action());
+		button3.addActionListener(new Action());
+		button4.addActionListener(new Action());
+		text.setHorizontalAlignment(NORMAL);
+
+		setDefaultCloseOperation(JFrame.EXIT_ON_CLOSE);
+	}
+
+	private class Action implements ActionListener {
+
+		@Override
+		public void actionPerformed(ActionEvent e) {
+
+			if (e.getSource() == button1) {
+				num = text.getText().indexOf("_");
+				line = text.getText().substring(0, num)
+						+ text.getText().substring(num + 1, num + 2)
+						+ "_"
+						+ text.getText().substring(num + 2,
```

Comment:
Aplikacija pada, excpetion je sljedeci:

Exception in thread "AWT-EventQueue-0" java.lang.StringIndexOutOfBoundsException: String index out of range: 2
	at java.lang.String.substring(Unknown Source)
	at ba.bitcamp.weekendhomework.GUI.tasks.Task4$Action.actionPerformed(Task4.java:61)

Koraci za reproduciranje:
1. Kliknuti na button4
2. Kliknuti na button1